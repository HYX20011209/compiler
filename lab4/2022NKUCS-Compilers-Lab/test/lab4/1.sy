int a;

int main()
{
    int hex = 0xa;
    int oct = 07;
    float f = 0.07;
    int a;
    a = 1 + 2;
    if(a < 5)
        return 1;
    return 0;
}