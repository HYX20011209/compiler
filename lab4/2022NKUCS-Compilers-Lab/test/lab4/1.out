Token			Lexeme			LineNum
INT				int				1
ID				a				1
SEMICOLON				;				1
INT				int				3
ID				main				3
LPAREN				(				3
RPAREN				)				3
LBRACE				{				4
INT				int				5
ID				hex				5
ASSIGN				=				5
HEX				10				5
SEMICOLON				;				5
INT				int				6
ID				oct				6
ASSIGN				=				6
OCT				7				6
SEMICOLON				;				6
FLOAT				float				7
ID				f				7
ASSIGN				=				7
FLT				0.07				7
SEMICOLON				;				7
INT				int				8
ID				a				8
SEMICOLON				;				8
ID				a				9
ASSIGN				=				9
DEC				1				9
ADD				+				9
DEC				2				9
SEMICOLON				;				9
IF				if				10
LPAREN				(				10
ID				a				10
ST				<				10
DEC				5				10
RPAREN				)				10
RETURN				return				11
DEC				1				11
SEMICOLON				;				11
RETURN				return				12
DEC				0				12
SEMICOLON				;				12
RBRACE				}				13
